hardhat.config.ts

```diff
@@ -52,11 +52,17 @@ const config: HardhatUserConfig = {
     },
   },
   solidity: {
-    compilers: [
-      {
-        version: '0.8.7',
+    version: '0.8.12',
+    settings: {
+      optimizer: {
+        enabled: true,
+        runs: 200,
+        details: {
+          yul: true,
+        },
       },
-    ],
+      viaIR: true,
```

Comment:
I suggest we set `viaIR` and `yul` to `false` so that it follows the same settings as PR 70. Does the code compile with these settings?

Reply:
It seems to be working with both on `true`, it is compiling and all tests are passing.